apps/studio/src/server/modules/folder/folder.router.ts

```diff
@@ -220,4 +265,58 @@ export const folderRouter = router({
         return pick(result, defaultFolderSelect)
       },
     ),
+
+  getIndexpage: protectedProcedure
+    .input(getIndexpageSchema)
+    .query(async ({ input: { resourceId, siteId } }) => {
+      // TODO: add permissions checking
+
+      return db.transaction().execute(async (tx) => {
+        const { title } = await tx
+          .selectFrom("Resource")
+          .where("id", "=", resourceId)
+          .select("title")
+          .executeTakeFirstOrThrow()
+
+        let indexPage = await tx
+          .selectFrom("Resource")
+          .where("Resource.parentId", "=", resourceId)
+          .where("Resource.type", "=", "IndexPage")
+          .select(["id", "draftBlobId"])
+          .executeTakeFirst()
+
+        // NOTE: Might want to create on click in
+        // rather than via the view
```

Comment:
issue: actually I do agree with this, feels a bit weird to me that the index page is created when you are browsing the folder with no intention of touching the index page.

suggestion: I think the `getIndexpage` can return `undefined` if the folder index page does not exist, then the frontend can call a separate mutation to create the index page on click before redirecting the user to that newly created index page. If already exists, we can maintain the current behaviour as a normal link.

Reply:
updated so that it throws an error, which we catch and then create the folder's index page. the initial decision was chosen because:

1. less waterfall timing (insteads of 2 requests, it's just one)
2. no need to handle loading states 

but given that they are separate responsibilities, it also makes sense to shift it out. i've went with this!

Reply:
Clarified offline, I think my main concern is just cos audit logs now log everything, so someone (like our team) can be casually browsing the site and inevitably created a new index page, which might raise questions from the agency if they scrutinise the audit logs in very great detail (but assumption here is that they actually do).

I think the other way of creating only when clicking into it won't really solve this issue anyway, so I'm ok to keep the current behaviour. We can revisit if really got people make a lot of noise about it (highly doubt they will).